src/components/RecipeForm.test.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import {fireEvent, render, screen} from '@testing-library/react';
+import RecipeForm from './RecipeForm';
+import {act} from "react-dom/test-utils";
+import userEvent from "@testing-library/user-event";
+
+describe('RecipeForm', () => {
+  describe('Given a recipe', () => {
+    const recipe = {
+      id: 1,
+      name: 'Pizza',
+      description: 'Put in the oven',
+      ingredients: [{
+        name: 'cheese'
+      },
+        {
+          name: 'mushrooms'
+        }]
+    }
+
+    it('should load successfully the form with the recipe values', async () => {
+      const onSubmitMock = jest.fn();
+      const onCancelMock = jest.fn();
+      render(<RecipeForm recipe={recipe} onSubmit={onSubmitMock} onCancel={onCancelMock}/>);
+
+      expect(screen.getByRole('button', {name: 'Save'})).toBeInTheDocument();
+      expect((screen.getByLabelText('Name') as HTMLInputElement).value).toBe('Pizza');
+      expect((screen.getByLabelText('Description') as HTMLInputElement).value).toBe('Put in the oven');
+      expect((screen.getByTestId('ingredient0') as HTMLInputElement).value).toBe('cheese');
+      expect((screen.getByTestId('ingredient1') as HTMLInputElement).value).toBe('mushrooms');
+    });
+
+    it('should submit the correct data when the user has removed an ingredient and clicks Save', async () => {
+      const onSubmitMock = jest.fn(e => e.preventDefault());
```

Comment:
are you sure you need to preventDefault on the mock function? I cannot remember doing it myslef

Reply:
Without doing that although the test still passes I get on the console the following error:

  console.error
    Error: Not implemented: HTMLFormElement.prototype.submit
        at module.exports (/home/user/Documents/Onboarding/exercises/reactExercise/react-recipe-exercise/app/node_modules/jsdom/lib/jsdom/browser/not-implemented.js:9:17)